src/prefect/engine.py

```diff
@@ -1466,6 +1480,23 @@ async def orchestrate_task_run(
                 )
                 args, kwargs = parameters_to_args_kwargs(task.fn, resolved_parameters)
 
+                # update task run name
+                if not run_name_set and task.task_run_name:
+                    sig = inspect.signature(task.fn)
+                    for kwarg, val in sig.parameters.items():
+                        if val.default is not val.empty:
+                            resolved_parameters.setdefault(kwarg, val.default)
+                    task_run_name = task.task_run_name.format(**resolved_parameters)
+                    await client.set_task_run_name(
+                        task_run_id=task_run.id, name=task_run_name
+                    )
+                    logger.extra["task_run_name"] = task_run_name
+                    logger.debug(
+                        f"Renamed task run {task_run.name!r} to {task_run_name!r}"
+                    )
+                    task_run.name = task_run_name
+                    run_name_set = True
```

Comment:
Since we resolve the parameters outside of while RUNNING block, can we just update the name once outside this and avoid tracking it with a boolean?

Reply:
I considered that and am open to it - the reason I did it this way was so that logs conformed to the new name